docs/onnx-conversion.md

```diff
@@ -1,75 +1,251 @@
 
+All scripts are available for reference under in the following directory:
+```
+src/main/python/onnx
+```
+
+The following tutorial will be using the scripts available in the above directory directly.
+
 ## Setups
-- pip requirements:
-    ```
-    onnx                          1.13.1
-    onnxoptimizer                 0.3.10
-    onnxruntime                   1.11.1
-    ```
-- SPLADE [Git repo](https://github.com/naver/splade):
-   ```bash
-   git clone https://github.com/naver/splade.git
-   ```
+
+pip requirements:
+```bash
+pip install torch transformers onnx onnxruntime onnxoptimizer
+```
 
 ## Converting from PyTorch models to ONNX model
 The following sections will describe how to convert SPLADE++ model to ONNX model. The steps are as follows:  
 
-### Load the corresponding HuggingFace model
+### Run the End to End PyTorch to ONNX Conversion
+Loading and running is done easily with argparse in the following script:
+```
+src/main/python/onnx/convert_hf_model_to_onnx.py
+```
+
+All that needs to be provided is the model_name as seen on huggingface. In our example we will be loading the SPLADE++ Ensemble Distil model found here:
+```
+naver/splade-cocondenser-ensembledistil
+```
+
+To run the script and produce the onnx model, run the following sequence of commands:
+```bash
+# Begin by going to the appropriate directory
+cd src/main/python/onnx
+# Now run the script
+python3 convert_hf_model_to_onnx.py --model_name naver/splade-cocondenser-ensembledistil
+```
+
+So what actually happens under the hood? The following sections will discuss the key parts of the above script:
+
+### Getting Output Specificaton from the Model
+
+Since we want our script to be generic, and be able to load any huggingface model with simplicity, we make it possible to fetch all required information from the model output structure with the following code:
+
 ```python
-from splade.models.transformer_rep import Splade
-import torch
+with torch.no_grad():
+    outputs = model(**test_input) # We provide the text test input and the model object.
 
-model_type_or_dir = "naver/splade-cocondenser-ensembledistil"
-model = Splade(model_type_or_dir, agg="max", fp16=True)
-model.eval()
+if isinstance(outputs, torch.Tensor):
+    return ['output']
+elif hasattr(outputs, 'last_hidden_state'):
+    return ['last_hidden_state']
+elif hasattr(outputs, 'logits'):
+    return ['logits']
+else:
+    return [f'output_{i}' for i in range(len(outputs))]
 ```
 
-### Creat dummy input variables to be used for ONNX conversion
+This is modularized in the function definition for ```get_model_output_names(model, test_input)```
+
+---
+
+Another important component is being able to specify the dimensions of the input and output tensors. This is achieved by the following code:
```

Comment:
I would say more that we are giving the dynamic axis names here and not specifying the dimensions